### PR TITLE
OAP-56 and logging

### PR DESCRIPTION
**Pull request by celinanperalta**
- Saves progress of web scraping incrementally using the `oapen_suggestions.endpoints` table in postgres
- Adds a base logger class for logging purposes, as well as a file handler for debug messages
- Adds tqdm progress bar to harvest job

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,9 +3,10 @@ services:
   oapen-engine :
     build: ./oapen-engine/
     environment:
-      - RUN_CLEAN=0
-      - COLLECTION_IMPORT_LIMIT=0 # Set to 0 for full harvest
-      - REFRESH_PERIOD=300
+      - RUN_CLEAN=1
+      - COLLECTION_IMPORT_LIMIT=0   # Set to 0 for full harvest
+      - REFRESH_PERIOD=86400        # daily
```
Comment (justinoboyle) [inline docker-compose.yml:8]: nice

File: oapen-engine/Dockerfile [hunk 2]
```diff
@@ -27,15 +28,26 @@ ENV PATH="$PATH:/usr/local/bin"
...
+RUN [ "python", "-c", "import nltk; nltk.download('punkt', download_dir='/home/appuser/nltk_data'); nltk.download('stopwords', download_dir='/home/appuser/nltk_data')" ]
+
 ENV PYTHONPATH="/home/appuser/src"
 
 # Install application into container
 COPY . .
 
+USER root
+
+RUN chmod u+x scripts/clean.sh
```
Comment (justinoboyle) [inline oapen-engine/Dockerfile:46]: do we want to refresh on every deploy?

Comment (celinanperalta) [reply oapen-engine/Dockerfile:46]: I added some scripts for manual refresh/suggestion generation, but there's a flag in docker-compose.yml to set whether or not to refresh on deploy

Comment (justinoboyle) [reply oapen-engine/Dockerfile]: oh awesome

**Review by justinoboyle — APPROVED** [review]

**Pull request merged**